pkg/scheduler/framework/interface.go

```diff
@@ -250,6 +250,9 @@ func NewStatus(code Code, reasons ...string) *Status {
 
 // AsStatus wraps an error in a Status.
 func AsStatus(err error) *Status {
+	if err == nil {
```

Comment:
why this check is needed?

Reply:
https://github.com/kubernetes/kubernetes/blob/5debf63a742fab8f3d9b3672a1fd3f14d2289dce/pkg/scheduler/schedule_one.go#L785-L788

We used to return an error here but now I changed to *framework.Status, like framework.AsStatus(err), but the problem here is err might be nil, so I added these codes. But I think the change makes sense and has no side effects.